components/authentication-framework/org.wso2.carbon.identity.application.authentication.framework/src/main/java/org/wso2/carbon/identity/application/authentication/framework/cache/AuthenticationContextCache.java

```diff
@@ -75,6 +82,16 @@ public static AuthenticationContextCache getInstance() {
      * @param entry Actual object where cache entry is placed.
      */
     public void addToCache(AuthenticationContextCacheKey key, AuthenticationContextCacheEntry entry) {
+
+        if (isSessionDataStorageOptimizationEnabled) {
+            try {
+                AuthenticationContextLoader.getInstance().optimizeAuthenticationContext(entry.getContext());
+            } catch (AuthenticationContextLoaderException e) {
+                log.debug(String.format("Error occurred while optimizing the Authentication context with " +
+                        "context id: %s", entry.getContext().getContextIdentifier()), e);
+                return;
```

Comment:
if an exception occurs, isn't it better to go with the normal flow?

Reply:
If there is a possibility of occurrence of an exception then by that time the authentication context would have been partially optimized, therefore in these conditions it is better not to follow the normal flow.